### PR TITLE
Mentor nerf

### PR DESCRIPTION
**Pull request by Tsurupeta**
Fixed mentors being able to see dsay, write in dsay, do stuff with tickets and kicking and much more that I've already forgotten.

## Changelog

:cl:
admin: fixed mentor perms
/:cl:

### CODE REVIEW

**Comment by Vizoplez** [comment]
May god bless you, my brother in christ.

**Review by Vizoplez — APPROVED** [review]

File: code/datums/communication/donator.dm [hunk 1]
```diff
@@ -12,9 +12,12 @@
...
 		return
 
-	if(!C.holder)
+	if(check_rights(R_INVESTIGATE))
+		return
+	else
 		if(!(C.donator_holder && C.donator_holder.flags & D_DOOC))
-			return FALSE
+			. = FALSE
+	return
```
Comment (ivanmixo) [inline code/datums/communication/donator.dm]: What the hell is this supposed to be?

`else` on an early return if statement? I don't get the `.` assignment either..

Comment (Tsurupeta) [reply code/datums/communication/donator.dm]: . = TRUE on first check. If `check_rights` returns true, the whole function returns TRUE and THE END.

I guess if you don't know the `.` operator and it's usage then it's kinda hard to read, so i'll change it to just returns

Comment (Tsurupeta) [reply code/datums/communication/donator.dm]: It already does the `. = ..()` and checks for `.` in the beginning, so...

**Review by Vivalas — APPROVED** [review]
tested, nice

**Pull request merged**